crates/sui-node/src/lib.rs

```diff
@@ -555,34 +542,61 @@ impl SuiNode {
         Self::start_epoch_specific_validator_components(
             config,
             state.clone(),
-            consensus_adapter,
+            connection_monitor_status,
             checkpoint_store,
             epoch_store,
             state_sync_handle,
             narwhal_manager,
             narwhal_epoch_data_remover,
-            validator_server_handle,
+            committee,
             accumulator,
             checkpoint_metrics,
             sui_tx_validator_metrics,
+            registry_service,
         )
         .await
     }
 
     async fn start_epoch_specific_validator_components(
         config: &NodeConfig,
         state: Arc<AuthorityState>,
-        consensus_adapter: Arc<ConsensusAdapter>,
+        connection_monitor_status: Arc<ConnectionMonitorStatus>,
         checkpoint_store: Arc<CheckpointStore>,
         epoch_store: Arc<AuthorityPerEpochStore>,
         state_sync_handle: state_sync::Handle,
         narwhal_manager: NarwhalManager,
         narwhal_epoch_data_remover: EpochDataRemover,
-        validator_server_handle: JoinHandle<Result<()>>,
+        committee: Arc<Committee>,
         accumulator: Arc<StateAccumulator>,
         checkpoint_metrics: Arc<CheckpointMetrics>,
         sui_tx_validator_metrics: Arc<SuiTxValidatorMetrics>,
+        registry_service: &RegistryService,
     ) -> Result<ValidatorComponents> {
+        let consensus_config = config
+            .consensus_config()
+            .ok_or_else(|| anyhow!("Validator is missing consensus config"))?;
+
+        // create a new map that gets injected into both the consensus handler and the consensus adapter
+        // the consensus handler will write values forwarded from consensus, and the consensus adapter
+        // will read the values to make decisions about which validator submits a transaction to consensus
+        let low_scoring_authorities = Arc::new(ArcSwap::new(Arc::new(HashMap::new())));
+
+        let consensus_adapter = Arc::new(Self::construct_consensus_adapter(
+            consensus_config,
+            state.name,
+            connection_monitor_status,
+            low_scoring_authorities.clone(),
+            &registry_service.default_registry(),
+        ));
+
+        let validator_server_handle = Self::start_grpc_validator_service(
+            config,
+            state.clone(),
+            consensus_adapter.clone(),
+            &registry_service.default_registry(),
+        )
+        .await?;
```

Comment:
Since this has been moved now under the `start_epoch_specific_validator_components` , will a new grpc server will always start ? Does the previous grpc server shutdown?

Reply:
I've reverted this to not start a new server every epoch, that was a bigger change than I was prepared to make here